spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BinderFactoryConfiguration.java

```diff
@@ -114,6 +117,13 @@ public BinderTypeRegistry binderTypeRegistry(ConfigurableApplicationContext conf
 		return new DefaultBinderTypeRegistry(binderTypes);
 	}
 
+	@Bean
+	@ConditionalOnEnabledHealthIndicator("binders")
```

Comment:
Added a note to expand our documentation to cover it https://github.com/spring-cloud/spring-cloud-stream/issues/278#issuecomment-177990814.